src/ResourceManager/OperationalInsights/Commands.OperationalInsights/Query/InvokeOperationalInsightsQuery.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.OperationalInsights.Models;
+using Microsoft.Azure.OperationalInsights;
+using Microsoft.Rest;
+using System;
+using System.Management.Automation;
+using System.Net;
+
+namespace Microsoft.Azure.Commands.OperationalInsights.Query
+{
+    [Cmdlet("Invoke", "AzureRmOperationalInsightsQuery"), OutputType(typeof(PSQueryResponse))]
```

Comment:
This should implement ShouldProcess: https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess

Reply:
That documentation seems to say ShouldProcess is only necessary if it changes the system, which our query API doesn't. Is there another reason we should support ShouldProcess?

Reply:
Does this cmdlet create a query which is saved anywhere in the system? The use of the verb "invoke" and the function being called "create" led me to believe the query was created and saved somewhere, in which case this is a cmdlet making changes to the system.  Let me know if this is incorrect in which case SupportsShouldProcess is not necessary.